test/unit/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -227,13 +227,21 @@ void _nrn_mechanism_register_data_fields(Args&&... args) {
             REQUIRE_THAT(generated,
                          ContainsSubstring(reindent_and_trim_text(expected_placeholder_nrn_state)));
         }
-        THEN("Placeholder registration function is printed") {
-            std::string expected_placeholder_reg = R"CODE(/** register channel with the simulator */
-    extern "C" void __test_reg() {
+        THEN("Initialization function for slist/dlist is printed correctly") {
+            std::string expected_initlists = R"(
+    static void _initlists() {
         /* s */
         _slist1[0] = {4, 0};
         /* Ds */
         _dlist1[0] = {7, 0};
+    };)";
```

Comment:
Why is there a semi-colon at the end of the function? More on point, could we check less? Just:
`"_slist1[0] = {4, 0}"` and `"_dlist1[0] = {7, 0}"`. I feel renaming `_initlists` should be permissible, and updating comments shouldn't require updating the tests.